### PR TITLE
Issue #58: Make built-in docs site reflect installation-specific config

### LINKED ISSUE #58: Built-in docs/help site
There should be a link in the header of every ESB page other than the kiosk views, that goes to a built-in /docs/ website. This should include similar documentation as the project’s github pages site, but with installation-specific values such as URLs, channel names, etc. interpolated into the documentation. It should also include an “About” page with a link to the equipment-status-board github project, the currently running version number, etc.

### PR DESCRIPTION
**Pull request by jantman**
## Summary

The built-in `/docs/` site (added in #60 for #58) still rendered generic placeholders — "the Equipment Status Board URL", "ask a staff member for the static status page URL", "the feature is only available when an administrator has set `ESB_BASE_URL`" — and documented features unconditionally even when they were disabled on a given deployment. This extends the existing placeholder mechanism with **real installation values** and **feature conditionals** so the live site tailors itself to what is actually configured, while the public GitHub Pages build stays a complete generic reference.

## New configuration

| Variable | Purpose |
|---|---|
| `STATIC_PAGE_PUBLIC_URL` | Public URL where members *view* the static status page (distinct from the existing push target). |
| `ORG_NAME` / `ORG_URL` / `ORG_BLURB` | De-hardcode "Decatur Makers" branding. Defaults preserve current rendering. |

All four are documented in `administrators.md` (env-var reference) and `.env.example`.

## Rendering engine (`esb/services/docs_service.py`)

- `get_placeholder_values()` now exposes installation values (`base_url`, `static_page_url`, `wifi_ssid`, `org_*`) and feature booleans (`qr_enabled`, `slack_enabled`, `static_page_enabled`, `wifi_configured`).
- The WiFi lookup (DB-backed `AppConfig`) **fails safe** so the public docs stay up on a fresh deployment before `flask db upgrade` — same resilience the admin guide already promises for `/metrics`.
- New `invalidate_page_cache()`, wired into the admin config-save path, so an edited WiFi SSID isn't served stale from the per-app render cache.

## Docs content

- Real URLs replace generic phrases; **Slack / QR / static-page / WiFi** sections are gated to the features actually enabled.
- The confusing "`ESB_BASE_URL` not configured" caveat now appears **only when QR is actually disabled**.
- Members QR section gains a "must be on WiFi" note (with the live SSID when set).
- **UI-accuracy fixes** surfaced by the audit: the create-repair entry point (the **Report Issue** button on an equipment page, not a non-existent "Repairs > New" menu) and the Admin tab navigation (Users/Areas/Config **tabs**, not a dropdown).

## Drift guard

Every new placeholder is mirrored in `mkdocs.yml` `extra:` with generic defaults; a new parity test asserts the runtime and mkdocs key sets stay identical; `mkdocs build --strict` produces no empty links.

## Testing

- `ruff check esb/ tests/` — clean
- `pytest tests/` — **1729 passed** (10 new docs tests: feature gating on/off, value substitution, branding override, WiFi cache invalidation, placeholder/mkdocs parity)
- `mkdocs build --strict` — builds clean, verified no empty links and that the public site shows the generic fallback text

Bumps version to `0.13.0` (minor — new feature), which will auto-cut a release on merge.

Closes #58.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/staff.md [hunk 1]
```diff
@@ -248,4 +249,4 @@ Click **Status** in the navigation bar to see the color-coded equipment grid org
 
 ### Static Status Page
 
-The static status page is an externally hosted lightweight version of the status dashboard. It is automatically regenerated and pushed whenever equipment status changes. This allows members to check status from outside the makerspace network. Configuration of the static page push method is handled by an administrator via environment variables.
+The static status page is an externally hosted lightweight version of the status dashboard. It is automatically regenerated and pushed whenever equipment status changes. This allows members to check status from outside the makerspace network. Configuration of the static page push method is handled by an administrator via environment variables.{% if static_page_url %} On this deployment it is published at [{{ static_page_url }}]({{ static_page_url }}) — share that link with members who need to check status remotely.{% endif %}
```
[the remark this answers is not in the document]
Comment (jantman) [reply docs/staff.md:253]: Good catch — fixed in fd16c1d. Wrapped the Staff guide's "Static Status Page" section in `{% if static_page_enabled %}` so it's omitted on live deployments without `STATIC_PAGE_PUBLIC_URL`, matching the documented behavior and the Members guide. The inner `{% if static_page_url %}` is retained so the public mkdocs build (which defaults `static_page_enabled: true` / `static_page_url: ''`) still renders the section without an empty link — verified via `mkdocs build --strict`.

File: tests/test_views/test_docs_views.py [hunk 2]
```diff
@@ -150,12 +150,152 @@ class TestDocsInterpolation:
...
+    def test_qr_section_hidden_when_base_url_unset(self):
+        assert 'Using QR Code Equipment Pages' not in _render('members', {})
+
+    def test_qr_section_shown_when_base_url_set(self):
+        assert 'Using QR Code Equipment Pages' in _render('members', QR_ON)
+
+    def test_static_page_section_gated_on_members(self):
+        assert 'Static Status Page' not in _render('members', {})
+        assert 'Static Status Page' in _render('members', STATIC_ON)
+
```
[the remark this answers is not in the document]
Comment (jantman) [reply tests/test_views/test_docs_views.py:210]: Done in fd16c1d — added `test_static_page_section_gated_on_staff` mirroring the Members assertion (absent with no config, present with `STATIC_PAGE_PUBLIC_URL` set). It now guards the previously-unconditional Staff section.

File: esb/services/docs_service.py [hunk 3]
```diff
@@ -94,11 +108,64 @@ def _pyproject_path():
...
+    Call after a docs-relevant ``AppConfig`` change (WiFi settings). Only the
+    ``('page', …)`` entries are dropped; the ``('meta', 'version')`` entry (env-
+    immutable) is left intact. Safe no-op if nothing has been cached yet.
     """
-    return {'oops_channel': current_app.config['SLACK_OOPS_CHANNEL']}
+    cache = current_app.extensions.get('docs_cache')
+    if not cache:
+        return
+    for key in [k for k in cache if k[0] == 'page']:
+        del cache[key]
```
[the remark this answers is not in the document]
Comment (jantman) [reply esb/services/docs_service.py]: Good catch — real race on the 2-thread gunicorn worker. Fixed in c4fdb43 two ways:

- `render_page()` (and `get_version()`) now read via the atomic `dict.get()` instead of check-then-index, so even an in-place deletion can't raise `KeyError` — a miss just re-renders.
- `invalidate_page_cache()` now replaces the cache dict atomically (`ext['docs_cache'] = {k: v ... if k[0] != 'page'}`) rather than `del`-ing keys in place, exactly as you suggested. A concurrent reader keeps its own reference to the old dict, and overlapping invalidations can no longer double-delete.

Added `test_invalidate_swaps_dict_leaving_old_intact` to lock in the swap-not-mutate contract. Full suite green (1731 passed).

File: mkdocs.yml [hunk 4]
```diff
@@ -33,9 +33,22 @@ plugins:
 
 # Generic default values rendered on the public GH Pages site where the docs
 # placeholders appear. Every key here MUST also exist in the docs service's
-# get_placeholder_values(); both renderers fail on drift.
+# get_placeholder_values(); both renderers fail on drift. The feature booleans
+# default to `true` so the public reference site documents every feature, even
+# though a given live deployment may have some disabled.
```
[the remark this answers is not in the document]
Comment (jantman) [reply mkdocs.yml]: Fixed in 63ccbaf. Reworded the comment to say *most* feature booleans default to `true`, and added an explicit note that `wifi_configured` defaults to `false` on purpose — with no real SSID to show, the public site renders the generic "be on the makerspace WiFi" wording instead of a placeholder network name.

File: docs/staff.md [hunk 5]
```diff
@@ -161,26 +161,26 @@ When someone scans a printed QR label, they see the public equipment status page
...
+Click **Admin** in the navigation bar, then the **Areas** tab, to manage the areas (rooms/zones) of the makerspace.
 
 ### Creating Areas
 
 1. Click **Add Area**
 2. Enter the area **name** (e.g., "Woodshop")
-3. Set the **Slack channel** for repair notifications for this area (e.g., `#woodshop-repairs`)
-4. Click Save
+{% if slack_enabled %}3. Set the **Slack channel** for repair notifications for this area (e.g., `#woodshop-repairs`)
+{% endif %}4. Click Save
```
[the remark this answers is not in the document]
Comment (jantman) [reply docs/staff.md]: Fixed in 1966a50. Switched the "Creating Areas" list to `1.` markers for every step, so the ordered list renumbers correctly in any Markdown renderer when the Slack-channel step is conditionally removed. Verified the rendered `<ol>` shows 1–4 with Slack enabled and 1–3 without.

File: esb/services/docs_service.py [hunk 6]
```diff
@@ -94,11 +108,69 @@ def _pyproject_path():
...
+    static_page_url = cfg.get('STATIC_PAGE_PUBLIC_URL', '')
+    # The docs site is public and unauthenticated; it must stay up even on a
+    # fresh deployment that has not yet run `flask db upgrade` (no app_config
+    # table) or during a transient DB outage. Treat any DB failure as "WiFi not
+    # configured" rather than 500-ing every guide page.
+    try:
+        wifi_ssid = config_service.get_config('wifi_ssid', '')
+    except SQLAlchemyError:
+        db.session.rollback()
+        wifi_ssid = ''
```
[the remark this answers is not in the document]
Comment (jantman) [reply esb/services/docs_service.py:147]: Good call — fixed in 1966a50. The WiFi-lookup fail-safe now mirrors metrics_service's rate-limited pattern: a full traceback on the first failure per process, then a one-line warning thereafter (`Failed to read wifi_ssid from AppConfig: ...`). So operators can spot a genuine DB outage while a pre-migration deployment serving /docs/ doesn't flood logs. Added a resilience test asserting a missing app_config table yields HTTP 200 and logs the warning.
